jcl/src/java.base/share/classes/java/lang/Class.java

```diff
@@ -4549,32 +4549,36 @@ private String getParameterTypesSignature(boolean throwException, String name, C
 	return signature.toString();
 }
 
+/*[IF JAVA_SPEC_VERSION <= 11] */
+private static Method getRootMethod;
 /*[IF JAVA_SPEC_VERSION == 8]*/
 /*[PR CMVC 114820, CMVC 115873, CMVC 116166] add reflection cache */
 private static Method copyMethod, copyField, copyConstructor;
+private static Field rootField;
 private static Field methodParameterTypesField;
 private static Field constructorParameterTypesField;
-private static final Object[] NoArgs = new Object[0];
 /*[ENDIF] JAVA_SPEC_VERSION == 8 */
+private static final Object[] NoArgs = new Object[0];
+/*[ENDIF] JAVA_SPEC_VERSION <= 11 */
 
 /*[PR JAZZ 107786] constructorParameterTypesField should be initialized regardless of reflectCacheEnabled or not */
 static void initCacheIds(boolean cacheEnabled, boolean cacheDebug) {
 	reflectCacheEnabled = cacheEnabled;
 	reflectCacheDebug = cacheDebug;
-	/*[IF JAVA_SPEC_VERSION == 8]*/
+	/*[IF JAVA_SPEC_VERSION <= 11]*/
 	AccessController.doPrivileged(new PrivilegedAction<Void>() {
 		@Override
 		public Void run() {
 			doInitCacheIds();
 			return null;
 		}
 	});
-	/*[ENDIF] JAVA_SPEC_VERSION == 8 */
+	/*[ENDIF] JAVA_SPEC_VERSION <= 11 */
 }
 static void setReflectCacheAppOnly(boolean cacheAppOnly) {
```

Comment:
While you're here, could you please add a blank line after line 4577, 4580 and line 4624 to separate the nearby methods?